Added dot net core support.

luwangshell opened the pull request:
New solution and project files for .net 5 are added to the repo.
Few updates to tweak the source code for .net core capabilities. For the ones which may not be compatible with both .net framework and .net core, added new files with Core suffix.

essenius commented on Rest/RestTests/XmlObjectTests.cs:
```diff
@@ -156,9 +157,13 @@ public void XmlObjectEvaluateTest()
+            //Assert.AreEqual("3349", xmlObject.Evaluate("/data/item[@id='myId']"));
+            Assert.AreEqual(0, string.Compare("3349", xmlObject.Evaluate("/data/item[@id='myId']"), CultureInfo.CurrentCulture, CompareOptions.IgnoreSymbols));
+            //Assert.AreEqual("2268", xmlObject.Evaluate("/data/item[@id='otherId']"));
+            Assert.AreEqual(0, string.Compare("2268", xmlObject.Evaluate("/data/item[@id='otherId']"), CultureInfo.CurrentCulture, CompareOptions.IgnoreSymbols));
```
Why the IgnoreSymbols? Please add a comment

luwangshell replied:
The ignoresysmbols is used here for cross platform support. e.g. /r/n for windows, /n for Mac. Will add a inline comment here too.

essenius commented on Rest/RestTests/RestRequestTests.cs:
```diff
@@ -40,9 +41,14 @@ public void RestRequestSetBodyForGetTest()
-            var target = new PrivateObject(restRequest);
-            target.Invoke("SetBody", "hello", Encoding.GetEncoding("iso-8859-1"), "GET");
-            var req = target.GetFieldOrProperty("_request") as HttpWebRequest;
+            // var target = new PrivateObject(restRequest);
```
Please remove all commented out code

essenius reviewed (changes requested):
See comments in the code.

luwangshell commented:
Hi @essenius the recommended changes are pushed to this branch. Also I have added a workflow for .net core ci. Currently only one test is failing on the typicode because the cookie is no longer returned. The build on .net core CI: https://github.com/essenius/FitNesseFitSharpRest/pull/2/checks?check_run_id=3233928681#step:6:128, is it okay we remove the cookie assertion part for now?

essenius commented:
@luwangshell OK to remove the cookie assertion

luwangshell commented:
Now both workflow are running fine.

luwangshell commented:
Have also updated the .net framework workflow to run all the unit tests as well.

luwangshell commented:
Hi @essenius , now i have updated the project and source code to support both net 5 and net 48. Also updated the CI workflow to accommodate the changes: https://github.com/essenius/FitNesseFitSharpRest/actions/runs/1103896968

essenius commented on Rest/RestTests/RestTests.csproj:
```diff
@@ -1,163 +1,33 @@
+ 
   <ItemGroup>
-    <Content Include="TestData.xml">
+    <None Update="TestData.xml">
```
If you moved the match parsing test data to code, this file should not be required anymore.

essenius commented on Rest/RestTests/SessionContextTests.cs:
```diff
@@ -37,27 +38,28 @@ public void SessionContextProxyTest()
             var context = new SessionContext();
             // We could do this via SetDefaults, but that is a bit elaborate. So cheating with a PrivateObject
-            var target = new PrivateObject(context);
+            // var target = new PrivateObject(context);
```
Remove commented out code

luwangshell replied:
fixed.

essenius commented on Rest/Rest/Rest.csproj:
```diff
@@ -1,119 +1,21 @@
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
+	<PackageReference Include="Microsoft.AspNetCore.Http" Version="2.2.2" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Add Json to the framework independent item group (same for both)

luwangshell replied:
fixed.

essenius commented on .github/workflows/fitsharp-rest-ci.yml:
```diff
@@ -0,0 +1,29 @@
+      run: dotnet build .\Rest\Rest.sln
+      
+    - name: Test
+      run: dotnet test .\Rest\Rest.sln
```
add newline at end of file (multiple places)

essenius commented on Rest/RestTests/ExtensionFunctionsTests.cs:
```diff
@@ -41,6 +41,7 @@ public void ExtensionFunctionsIsLikeTest()
             Assert.IsTrue("abc".IsLike("a*"));
         }
 
+#if NET48
```
We should be able to take the same approach for NET48 and NET50, and eliminate the need for TestData.xml

essenius reviewed (commented):
Left a couple of comments in the code.
